### DISCUSSION TITLE
Am I able to config Marksman without a `.toml` file?

### DISCUSSION THREAD

**Discussion by ghost** [artempyanykh/marksman#248] [Q&A]
Hi all, 

I'm using Marksman in Neovim, just making two changes to the defaults, and I'd like to configure it via `nvim-lspconfig` options, as that's how I've configured some Python-related LSPs ([example](https://www.lazyvim.org/extras/lang/python#nvim-lspconfig)). **Is this possible?** I don't use the LSP outside nvim, and my neurosis would *love* to set it up within the rest of my LSP config instead of using this loose `toml` file.

I posted this question over on the [LazyVim](https://github.com/LazyVim/LazyVim/discussions/1442) repo as well, thinking I was just being dumb with the syntax, but another user suggested Marksman might not even support what I'm trying to do. 

Any advice is appreciated!

**Comment by artempyanykh** [comment]
Hi @SaltGolem! Thanks for raising this. Currently, marksman allows only minimal configuration over LSP (only text sync kind). I think it's a good idea to extend this and allow providing full config as part of lsp init request.

With that said, I don't have the cycles to work on this myself, but will happily accept PRs :)
